### PR TITLE
fix ios inputOnToolbar missing cursor

### PR DESCRIPTION
**Pull request by novain1102**
Re: https://github.com/cocos/3d-tasks/issues/16114#
### Changelog
- fix ios inputOnToolbar cursor missing 
- revise some codes and add back missing codes in Editbox-ios.mm

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/ui/edit-box/EditBox-ios.mm [hunk 1]
```diff
@@ -564,14 +566,22 @@ - (void) show: (const cc::EditBox::ShowInfo*)showInfo {
...
+     dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.1 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+         if(![[curView inputOnToolbar] becomeFirstResponder]) {
+             CC_LOG_ERROR("inputOnToolbar becomeFirstResponder error!");
+         }
+    });
+
 }
 // Change the focus point to the TextField or TextView on the toolbar.
 - (void) hide {
-    [[curView inputOnView] becomeFirstResponder];
```
Comment (novain1102) [inline native/cocos/ui/edit-box/EditBox-ios.mm:571]: There's no need to set inputOnView `becomeFirstResponder`. It'll waste time

File: native/cocos/ui/edit-box/EditBox-ios.mm [hunk 2]
```diff
@@ -564,14 +566,22 @@ - (void) show: (const cc::EditBox::ShowInfo*)showInfo {
...
-    [[curView inputOnView] removeFromSuperview];
-    [[curView inputOnToolbar] resignFirstResponder];
-    [[curView inputOnView] resignFirstResponder];
+   if ([[curView inputOnToolbar] isFirstResponder]) {
```
Comment (novain1102) [inline native/cocos/ui/edit-box/EditBox-ios.mm:578]: use `isFirstResponder` is more efficient

File: native/cocos/ui/edit-box/EditBox-ios.mm [hunk 3]
```diff
@@ -95,6 +95,7 @@ - (void)        textViewDidChange:(UITextView *)textView;
  ************************************************************************/
 namespace {
 static bool g_isMultiline{false};
+static bool g_confirmHold{false};
```
Comment (novain1102) [inline native/cocos/ui/edit-box/EditBox-ios.mm:98]: Seems lost this attribute when refactor editbox.

File: native/cocos/ui/edit-box/EditBox-ios.mm [hunk 4]
```diff
@@ -607,12 +618,12 @@ - (IBAction)buttonTapped:(UIButton *)button {
 
 void EditBox::hide() {
     [[EditboxManager sharedInstance] hide];
-    EditBox::_isShown = true;
+    EditBox::_isShown = false;
```
Comment (novain1102) [inline native/cocos/ui/edit-box/EditBox-ios.mm:621]: logic error before

**Review by minggo — APPROVED** [review]

**Pull request merged**